### PR TITLE
fix: fixed __next__ method

### LINKED ISSUE #38: cursor iterator is infinite
in a given below example cursor iterator is infinite, is it expected behaviour?
```
from sqlalchemy import create_engine, text
engine = create_engine(
    f"databend://{username}:{password}@{host_port_name}/{database_name}?sslmode=disable"
)
connection = engine.connect()
cursor = connection.execute(text("SELECT 1")).cursor
for row in cursor:
    print(row)
```

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
Fixes #38

### CODE REVIEW

**Review by everpcpc — APPROVED** [review]

File: databend_sqlalchemy/connector.py [hunk 1]
```diff
@@ -276,9 +276,9 @@ def __next__(self):
         as :py:meth:`fetchone`. A ``StopIteration`` exception is raised when the result set is
         exhausted.
         """
-        return self.fetchone()
+        return self._rows.__next__
```
Comment (everpcpc) [inline databend_sqlalchemy/connector.py:279]: should check if self._rows is not None?

Comment (AndrewChubatiuk) [reply databend_sqlalchemy/connector.py]: added condition

**Review by everpcpc — APPROVED** [review]

**Comment by AndrewChubatiuk** [comment]
@everpcpc could you please release this fix?
going to use it in https://github.com/getredash/redash/pull/6671

**Comment by everpcpc** [comment]
cc @hantmac could you help release this fix?

**Comment by hantmac** [comment]
> @everpcpc could you please release this fix? going to use it in [getredash/redash#6671](https://github.com/getredash/redash/pull/6671)

@AndrewChubatiuk Hi we have released the 0.4.6 version , https://pypi.org/manage/project/databend-sqlalchemy/release/0.4.6/

**Pull request merged**